.github/workflows/ci.yml

```diff
@@ -0,0 +1,27 @@
+name: run tests 
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+    unit-tests: 
+      runs-on: ubuntu-latest
+
+      steps: 
+        - name: Checkout code
+          uses: actions/checkout@v4
+
+        - name: Set up environment 
+          uses: actions/setup-node@v4
+          with: 
+            node-version: '20'
+
+        - name: Install dependencies
+          run: npm install
+
+        - name: Run tests
+          run: npx vitest
```

Comment:
this is already a command in package.json, so we can just do `npm test` (or `pnpm test`)

Reply:
:+1: